deps/recastnavigation/Detour/Source/DetourCommon.cpp

```diff
@@ -226,10 +226,18 @@ bool dtClosestHeightPointTriangle(const float* p, const float* a, const float* b
 
 	// If point lies inside the triangle, return interpolated ycoord.
 	if (u >= 0.0f && v >= 0.0f && (u + v) <= denom) {
-		h = a[1] + (v0[1] * u + v1[1] * v) / denom;
-		return true;
-	}
-	return false;
+
+        if (denom == 0.f && u == 0.f && v == 0.f)
```

Comment:
Is it a custom change on the recastnavigation library, or am I wrong?

Reply:
Yes, one of Trinity's commits, due to some pathing issues in ICC (e.g Professor).

Shall we leave a comment somewhere?

Reply:
if it's not a "recastnavigation" official code it's strongly suggested to leave at least a comment here. However, it's very risky since if we want to update the library in future we might lost this change then.

Reply:
They made their own fork here https://github.com/TrinityCore/recastnavigation/tree/3.3.5, maybe we can do the same and use it instead by linking it as a git repo within deps?

Reply:
I'm going to setup a proper monorepo strategy for it using git subrepo/subtree, thus we can safely edit deps without worrying about losing changes after an update.

I'm also working to a parallel solution for sloped angles that is done directly at runtime instead of requiring the re-extraction of maps. We can then check if it works together with this PR and eventually clean up a little bit the code.

Please, do not merge this PR until we will have a proper solution for the deps custom changes. 

Stay tuned.